docs/projects-to-work-on.md

```diff
@@ -0,0 +1,128 @@
+
+- **Mainlining more patches to upstream GNAT:** GNAT/GCC-Ada is a wonderful
+  piece of technology, and thanks to it being based on the GCC compiler
+  infraestructure, it can target a large number of Operating Systems and
+  architectures. However, GNAT/GCC-Ada requires a bit of setup and additional
+  files that are needed in order to compile Ada code for these targets. The most
+  common architecture/OS pairs are already present inside GCC, but some
+  communities maintain their own set of patches in order to support GNAT. It
+  would be ideal if those patches could become part of the GCC source tree. Here
+  are some resources and patches that would need mainlining:
+  - **[Android](https://source.android.com/):** in the past there were some
+    versions of GNAT that had support for Android. Some patches for Android
+    support can be found in [this
+    repository](https://github.com/search?q=repo%3Ajrmarino%2Fdraco%20path%3A%2F%5Emisc%5C%2Fflux13%5C%2F%2F%20android&type=code). Nontheless, Ada also supports Android with WASM, [see this demo](https://blog.adacore.com/android-application-with-ada-and-webassembly).
+  - **[OpenBSD](https://www.openbsd.org/):** OpenBSD has supported GNAT for a
+    few architectures for a long while, however, the patches to add support have
+    never been mainlined. The patches can be found in their ports tree or in
+    [this Github
+    mirror](https://github.com/openbsd/ports/tree/master/lang/gcc/11/patches)
+  - **[NetBSD](http://netbsd.org/):** updated GCC patches to add support for
+    NetBSD (amd64) can be found
+    [here](https://github.com/jrmarino/draco/tree/master/misc/flux13). Adding
+    support for other architectures can somewhat easily be done. Thank to
+    J. Marino for his work.
+  - **[FreeBSD](https://www.freebsd.org/):** FreeBSD support within GNAT could
+    use quite a bit of help. This is [the current patch used by
+    FreeBSD](https://gitlab.com/FreeBSD/freebsd-ports/-/tree/main/lang/gnat12/files).
+- **Adding support for more OSes:** support could be added to the following
+  Operating Systems:
+  - **[Zephyr](https://www.zephyrproject.org/):** it is an RTOS that has been
+    gaining a lot of traction and support. Initial support for the Zephyr RTOS
+    was started [here](https://github.com/ila-embsys/zephyr-ada-gnat-rts).
+  - **[Ironclad](https://ironclad.cx/):** it is a POSIX compatible RTOS
+    completely written in Ada. It currently has no RTS (RunTime System) support
+    for compiling Ada programs targetting it.
+  - **[Illumos](https://www.illumos.org/):** Illumos is the continuation of
+    OpenSolaris as an open source project. It currently has no Ada support
+    within GNAT.
+- **[GNAT-LLVM](https://github.com/AdaCore/gnat-llvm/):** the GNAT-LLVM project is fairly new but very promissing. It would be nice to test it, help with the documentation and create cool demos, see [this Ada-WASM example](https://github.com/godunko/adawebpack).
```

Comment:
```suggestion
- **[GNAT-LLVM](https://github.com/AdaCore/gnat-llvm/):** the GNAT-LLVM project is fairly new but very promising. It would be nice to test it, help with the documentation and create cool demos, see [this Ada-WASM example](https://github.com/godunko/adawebpack).
```